plugins/system/cookieconsent/cookieconsent.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  System.cookieconsent
+ *
+ * @copyright   Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Language\Associations;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Router\Route;
+
+
+/**
+ * Cookie consent plugin to add simple cookie information.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class PlgSystemCookieconsent extends CMSPlugin
+{
+	/**
+	 * If true, language files will be loaded automatically.
+	 *
+	 * @var    boolean
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $autoloadLanguage = true;
+
+ 	/**
+	 * Application object.
+	 *
+	 * @var    JApplicationCms
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $app;
+
+	/**
+	 * Database object.
+	 *
+	 * @var    JDatabaseDriver
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $db;
+
+	/**
+	 * Return the url of the assigned article based on the current user language
+	 *
+	 * @return  string  Returns the link to the cookie policy
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function getAssignedPolicylinkUrl()
+	{
+		$db = Factory::getDbo();
```

Comment:
```suggestion
		$db = $this->db;
```
And remove `use Joomla\CMS\Factory;`